src/GraphQL/ObjectExtensions.cs

```diff
@@ -296,25 +296,31 @@ public static T GetPropertyValue<T>(this object value)
         /// <param name="value">The value being tested.</param>
         public static bool IsDefinedEnumValue(Type type, object value)
         {
-            var names = Enum.GetNames(type);
-            if (names.Contains(value?.ToString() ?? "", StringComparer.OrdinalIgnoreCase))
+            try
             {
-                return true;
-            }
+                var names = Enum.GetNames(type);
+                if (names.Contains(value?.ToString() ?? "", StringComparer.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
 
-            var underlyingType = Enum.GetUnderlyingType(type);
-            var converted = Convert.ChangeType(value, underlyingType);
+                var underlyingType = Enum.GetUnderlyingType(type);
+                var converted = Convert.ChangeType(value, underlyingType);
 
-            var values = Enum.GetValues(type);
+                var values = Enum.GetValues(type);
 
-            foreach (var val in values)
-            {
-                var convertedVal = Convert.ChangeType(val, underlyingType);
-                if (convertedVal.Equals(converted))
+                foreach (var val in values)
                 {
-                    return true;
+                    var convertedVal = Convert.ChangeType(val, underlyingType);
+                    if (convertedVal.Equals(converted))
+                    {
+                        return true;
+                    }
                 }
             }
+            catch
+            {
+            }
```

Comment:
I added a try..catch block so that this function returns false when an exception is thrown.